File: SEMESTER 2/Programming/End Semester Solvents/3_bubble_sort.c
// CODE 3: Write a program to implement a sort technique that works by repeatedly stepping through the list to be sorted.

#include <stdio.h>
#define swap(a, b) a ^= b, b ^= a, a ^= b
int size;

int bsort(int *arr)
{
    int count = 0;
    for (int i = 0; i < size; i++)
        for (int j = i; j < size; j++)
            if (arr[i] > arr[j])
                count++,
                    swap(arr[i], arr[j]);

    return count;
}

void print(int *array)
{
    for (int i = 0; i < size - 1; i++)
        printf("%i, ", array[i]);

    printf("%i\n", array[size - 1]);
}

void main()
{
    int arr[] = {7, 1, 4, 12, 67, 33, 45};
    size = sizeof arr / sizeof arr[0];

    int swap = bsort(arr);
    print(arr);
    printf("swaps: %i", swap);
}

File: SEMESTER 2/Programming/End Semester Solvents/3_bubble_sort.py
# CODE 3: Write a program to implement bubble sort. Given the numbers 7, 1, 4, 12, 67, 33, and 45. How many swaps will be performed to sort these numbers using the bubble sort.


def bsort(arr: list[int]) -> int:
    count: int = 0
    for i in range(len(arr)):
        for j in range(i, len(arr)):
            if arr[i] > arr[j]:
                arr[i], arr[j] = arr[j], arr[i]
                count += 1
    return count


a = [7, 1, 4, 12, 67, 33, 45]
swap = bsort(a)
print(*a, sep=", ")
print(f"swaps: {swap}")
